roles/cockpit-ui/tasks/main.yml

```diff
@@ -22,43 +22,45 @@
   changed_when: false
 
 - name: Create passthrough route for docker-registry
-  command: >
-    {{ openshift.common.client_binary }} create route passthrough
-    --service docker-registry
-    --config={{ openshift_hosted_kubeconfig }}
-    -n default
-  register: create_docker_registry_route
-  changed_when: "'already exists' not in create_docker_registry_route.stderr"
-  failed_when: "'already exists' not in create_docker_registry_route.stderr and create_docker_registry_route.rc != 0"
+  oc_route:
+    name: passthrough
+    namespace: default
+    service_name: docker-registry
+    kubeconfig: "{{ openshift_hosted_kubeconfig }}"
+    state: present
+  run_once: true
   when: check_docker_registry_exists.rc == 0
 
 - name: Create passthrough route for registry-console
-  command: >
-    {{ openshift.common.client_binary }} create route passthrough
-    --service registry-console
-    --port registry-console
-    --config={{ openshift_hosted_kubeconfig }}
-    -n default
-  register: create_registry_console_route
-  changed_when: "'already exists' not in create_registry_console_route.stderr"
-  failed_when: "'already exists' not in create_registry_console_route.stderr and create_registry_console_route.rc != 0"
+  oc_route:
+    name: passthrough
+    namespace: default
+    service_name: registry-console
+    kubeconfig: "{{ openshift_hosted_kubeconfig }}"
+    state: present
+  run_once: true
   when: check_docker_registry_exists.rc == 0
 
 - name: Retrieve docker-registry route
-  command: >
-    {{ openshift.common.client_binary }} get route docker-registry
-    -o jsonpath='{.spec.host}'
-    --config={{ openshift_hosted_kubeconfig }}
-    -n default
+  oc_route:
+    name: docker-registry
+    namespace: default
+    kubeconfig: "{{ openshift_hosted_kubeconfig }}"
+    key_path:
+      jsonpath: "{.spec.host}"
+    state: list
```

Comment:
Having to run the module a second time to retrieve info about the route seems wrong. I would expect the output of oc_route with state=present to return the created or existing route back.